src/core/lib/event_engine/windows/iocp.h

```diff
@@ -0,0 +1,67 @@
+// Copyright 2022 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef GRPC_CORE_LIB_EVENT_ENGINE_WINDOWS_IOCP_H
+#define GRPC_CORE_LIB_EVENT_ENGINE_WINDOWS_IOCP_H
+
+#include <grpc/support/port_platform.h>
+
+#ifdef GPR_WINDOWS
+
+#include "absl/status/status.h"
+
+#include <grpc/event_engine/event_engine.h>
+
+#include "src/core/lib/event_engine/poller.h"
+#include "src/core/lib/event_engine/windows/socket.h"
+
+namespace grpc_event_engine {
+namespace experimental {
+
+class IOCP final : public Poller {
+ public:
+  explicit IOCP(EventEngine* event_engine) noexcept;
```

Comment:
Having to pass an entire `EventEngine` in here makes it harder to test this in isolation. Is there a more restrictive interface we can pass here, and offer a fake in tests?

Reply:
I extracted an Executor interface, it's a much simpler thing now.